kore/src/Kore/Log/DebugAppliedRewriteRules.hs

```diff
@@ -62,6 +62,10 @@ instance Pretty DebugAppliedRewriteRules where
 instance Entry DebugAppliedRewriteRules where
     entrySeverity _ = Debug
     helpDoc _ = "log applied rewrite rules"
+
+    isEmpty DebugAppliedRewriteRules{appliedRewriteRules}
+        | null appliedRewriteRules = True
+        | otherwise = False
```

Comment:
There is now no way (I think) to filter the message once we already start converting it to `LogStr`, so we would either have to render it twice, once in the filter (where we could then check if it is empty/null) and second time when we actually want to log it. As this would be inefficient, I've added a type-class function which can be used to control if some kore Entry is just used for context and should thus not be emitted as a message. See occurrences of `isEmpty` in this PR to see what I mean.